Source/WebCore/workers/WorkerInspectorProxy.cpp

```diff
@@ -35,24 +35,90 @@
 #include <JavaScriptCore/InspectorAgentBase.h>
 #include <wtf/NeverDestroyed.h>
 #include <wtf/TZoneMallocInlines.h>
+#include <wtf/WeakHashSet.h>
 
 namespace WebCore {
 
 WTF_MAKE_TZONE_ALLOCATED_IMPL(WorkerInspectorProxy);
 
 using namespace Inspector;
 
-static Lock proxiesLock;
-static WeakHashSet<WorkerInspectorProxy>& proxies() WTF_REQUIRES_LOCK(proxiesLock)
+static Lock proxiesPerWorkerGlobalScopeLock;
+static HashMap<ScriptExecutionContextIdentifier, WeakHashSet<WorkerInspectorProxy>>& proxiesPerWorkerGlobalScope() WTF_REQUIRES_LOCK(proxiesPerWorkerGlobalScopeLock)
 {
-    static NeverDestroyed<WeakHashSet<WorkerInspectorProxy>> proxies;
+    static NeverDestroyed<HashMap<ScriptExecutionContextIdentifier, WeakHashSet<WorkerInspectorProxy>>> proxies;
     return proxies;
 }
 
-WeakHashSet<WorkerInspectorProxy> WorkerInspectorProxy::allWorkerInspectorProxiesCopy()
+static HashMap<PageIdentifier, WeakHashSet<WorkerInspectorProxy>>& proxiesPerPage()
 {
-    Locker lock { proxiesLock };
-    return proxies();
+    static MainThreadNeverDestroyed<HashMap<PageIdentifier, WeakHashSet<WorkerInspectorProxy>>> proxies;
+    return proxies;
+}
+
+void WorkerInspectorProxy::addToProxyMap()
```

Comment:
NIT: is there any benefit to having this be a member function instead of a static method?

Reply:
Then we'd need to pass 2 parameters: `m_contextIdentifier` and `this`. I figured that if I have to pass `this`, I might as well make it a member function.